src/xtgeo/grid3d/_gridprop_import_xtgcpprop.py

```diff
@@ -23,21 +37,25 @@ def import_xtgcpprop(mfile, ijrange=None, zerobased=False):
 
     """
     offset = 36
-    with open(mfile.file, "rb") as fhandle:
-        buf = fhandle.read(offset)
 
-    # unpack header
-    swap, magic, nbyte, ncol, nrow, nlay = unpack("= i i i q q q", buf)
+    if not isinstance(mfile.file, Path):
```

Comment:
Im not really sure what todo in this case, i wonder if we should add a file-reader/opener context manger to the `_XTGeoFile` class that provides a filelike interface when reading.

Reply:
I do think a context manager for header reads would be valuable generally, though